utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -242,36 +242,83 @@ impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
 }
 
 impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
-    /// Generates a function which calls "properly" the `gr_reservation_send`.
-    fn generate_send_from_reservation(&mut self) {
-        let (reserve_gas_idx, reservation_send_idx) = {
-            let maybe_reserve_gas = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReserveGas))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-            let maybe_reservation_send = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReservationSend))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-
-            match maybe_reserve_gas.zip(maybe_reservation_send) {
-                Some(indexes) => indexes,
-                None => {
-                    log::trace!("Wasm hasn't got either gr_reserve_gas, or gr_reservation_send, or both of them");
-                    return;
-                }
-            }
-        };
+    /// Returns the indexes of invocable sys-calls.
+    fn invocable_sys_calls_indexes(&self, sys_calls: &[SysCallName]) -> Option<Vec<usize>> {
+        let ret: Option<Vec<_>> = sys_calls
+            .iter()
+            .map(|&sys_call| {
+                self.sys_calls_imports
+                    .get(&InvocableSysCall::Loose(sys_call))
+                    .map(|&(_, call_indexes_handle)| call_indexes_handle)
+            })
+            .collect();
+
+        if ret.is_none() {
+            log::trace!(
+                "Wasm should import the following syscalls: {missing_sys_calls}",
+                missing_sys_calls = sys_calls
+                    .iter()
+                    .map(|sys_call| sys_call.to_str())
+                    .collect::<Vec<_>>()
```

Comment:
collect to an array of known size?

Reply:
rust has no way to collect into an array in stdlib

Reply:
Yeah, but, I mean we should avoid allocating. Maybe introduce some external converter function or something similar

Reply:
done in 6aa698f1f6f75fffd984f8773d5acc4345da14e1